src/main/java/org/spongepowered/common/SpongeCommon.java

```diff
@@ -100,8 +102,21 @@ public static RegistryAccess.Frozen vanillaRegistryAccess() {
         return SpongeCommon.server().registryAccess();
     }
 
+    public static SpongeRegistryHolder scopedHolder() {
+        var holder = PhaseTracker.getInstance().currentCause().first(SpongeRegistryHolder.class);
```

Comment:
I still strongly feel that this is violating Cause. We need a static area to store this field, separate from an engine per my original idea. So, it is perfectly fine to put it on the `PhaseTracker` as an optional field. But not in the Cause. Cause is meant to be inspected, and plugin developers are told to take it as face value. Not for us to use it as a dumping ground.

Reply:
Not to mention, you are creating frames where they were not needed before. If you put the field on the tracker directly, no need for these pointless frames.

Reply:
the frames are created 6 times

Reply:
But I guess now that I realized I never needed to push the Server at `#initServer` I could just put a try finally block around there